### PR TITLE
ESM tests

### PR DESCRIPTION
**Pull request by grigasp**
Closes https://github.com/iTwin/presentation/issues/713

### CODE REVIEW

File: apps/full-stack-tests/src/RenderUtils.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+function createDefaultWrappers(disableStrictMode?: boolean, addThemeProvider?: boolean) {
+  const wrappers: Array<ReturnType<typeof createWrapper>> = [];
+  if (addThemeProvider) {
+    wrappers.push(createWrapper(ThemeProvider));
+  }
+
+  // if `DISABLE_STRICT_MODE` is set do not wrap components into `StrictMode` component
+  const StrictModeWrapper = process.env.DISABLE_STRICT_MODE || disableStrictMode ? Fragment : StrictMode;
```
Comment (saskliutas) [inline apps/full-stack-tests/src/RenderUtils.tsx:32]: We don't need add StrictMode wrapper anymore. RTL allows enabling strict mode. e.g. https://github.com/iTwin/presentation/blob/7f354909f0a538c20622d320cc2f34f4eb2c0359/packages/hierarchies-react/src/test/TestUtils.ts#L22

File: packages/components/src/test/setup.ts [hunk 2]
```diff
@@ -31,7 +31,7 @@ global.ResizeObserver = class ResizeObserver {
 
 // supply mocha hooks
 import path from "path";
-import { cleanup } from "@testing-library/react";
+const { cleanup } = await import("@testing-library/react");
```
[the remark this answers is not in the document]
Comment (grigasp) [reply packages/components/src/test/setup.ts:34]: With ESM all the `import` directives are executed as soon as the module is loaded, without executing prior code first. This means that `@testing-library/react` was loaded before JSDOM was injected. Testing library, in turn, caused React to be loaded, and there was no JSDOM at that point, making React misbehave. Spent half a day on this...

File: packages/test-utilities/node-hooks/ignore-styles.cjs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+/*---------------------------------------------------------------------------------------------
```
[the remark this answers is not in the document]
Comment (grigasp) [reply packages/test-utilities/node-hooks/ignore-styles.cjs:1]: No specific reason. Do you see any issues with that?

File: apps/full-stack-tests/src/RenderUtils.tsx (hunk 1 above)
Comment (grigasp) [reply apps/full-stack-tests/src/RenderUtils.tsx]: Cool, removed the wrapper from components and full stack tests.

**Pull request merged**